plugins/history3/databasemanager.cpp

```diff
@@ -0,0 +1,138 @@
+		db.exec(
+					"CREATE TABLE IF NOT EXISTS \"messages\" ("
+					"\"id\" Integer Primary Key Autoincrement Not Null, "
+					"\"timestamp\" Integer, "
+					"\"message\" Text, "
+					"\"protocol\" Text Not Null, "
+					"\"account\" Text Not Null, "
+					"\"direction\" Integer Not Null, "
+					"\"importance\" Integer, "
+					"\"contact\" Text, "
+					"\"subject\" Text, "
+					"\"session\" Text, "
+					"\"session_name\" Text, "
+					"\"from\" Text, "
+					"\"from_name\" Text, "
+					"\"to\" Text, "
+					"\"to_name\" Text, "
+					"\"state\" Integer, "
+					"\"type\" Integer, "
+					"\"is_group\" Integer Default'0') "
+					);
+	}
+}
+
+void DatabaseManager::insertMessage(Kopete::Message &message)
+{
+	QSqlQuery query(db);
+
+	//Prepare an SQL query to insert the message to the db
+	query.prepare("INSERT INTO `messages` (`timestamp`, `message`, `account`, `protocol`, `direction`, `importance`, `contact`, `subject`, "
+		      " `session`, `session_name`, `from`, `from_name`, `to`, `to_name`, `state`, `type`, `is_group`) "
+		      " VALUES (:timestamp, :message, :account, :protocol, :direction, :importance, :contact, :subject, :session, :session_name, "
+		      " :from, :from_name, :to, :to_name, :state, :type, :is_group)");
+
+	//Add the values to the database fields.
+	query.bindValue(":timestamp", QString::number(message.timestamp().toTime_t()));
+	query.bindValue(":message", message.parsedBody());
+	query.bindValue(":account", message.manager()->account()->accountId());
+	query.bindValue(":protocol", message.manager()->account()->protocol()->pluginId());
+	query.bindValue(":direction", QString::number(message.direction()));
+	query.bindValue(":importance", QString::number(message.importance()));
+
+	QString contact;
+	if (message.direction() == Kopete::Message::Outbound) {
+		contact = message.to().at(0)->contactId();
+	} else {
+		contact = message.from()->contactId();
+	}
+	query.bindValue(":contact", contact);
+
+	query.bindValue(":subject", message.subject());
+	query.bindValue(":session", "");
+	query.bindValue(":session_name", "");
+	query.bindValue(":from", message.from()->contactId());
+	query.bindValue(":from_name", message.from()->displayName());
+
+	//If we are dealing with only one recepient, we save this as a single user chat
+	if (message.to().count() == 1) {
+		query.bindValue(":to", message.to().at(0)->contactId());
+		query.bindValue(":to_name", message.to().at(0)->displayName());
+		query.bindValue(":is_group", "0");
+	} else {
+		//Otherwise we will create a string with the contact ids of the recepients, and another string to
+		//hold the contact names of the recepients. Both these strings are comma delimited.
+		QString to, to_name;
+		foreach (Kopete::Contact *c, message.to()) {
+			to.append(c->contactId() + ",");
+			to_name.append(c->displayName() + ",");
+		}
+		to.chop(1);
+		to_name.chop(1);
+		query.bindValue(":to", to);
+		query.bindValue(":to_name", to_name);
+		query.bindValue(":is_group", "1");
```

Comment:
I think it would make sense to move the strings to `DatabaseConstants` class and have their static initialization there.